src/npc/ai/hell.rs

```diff
@@ -72,4 +76,808 @@ impl NPC {
+        match self.action_num {
+            0 => {
+                self.npc_flags.set_shootable(false);
+                self.npc_flags.set_ignore_solidity(false);
+                self.damage = 0;
+                self.action_num = 1;
+                self.anim_num = 0;
+                self.display_bounds.top = 0x1800;
+                self.display_bounds.right = 0x1800;
+                self.display_bounds.left = 0x1800;
+                self.vel_y = -0x200;
+
+                match self.direction {
+                    Direction::Left => self.vel_x = 0x100,
+                    Direction::Right => self.vel_x = -0x100,
+                    _ => (),
+                };
+                state.sound_manager.play_sfx(50);
+            }
+            1 if self.flags.hit_bottom_wall() => {
+                self.action_num = 2;
+                self.action_counter = 0;
+                self.anim_num = 1;
+                self.anim_counter = 0;
+            }
+            2 => {
+                self.vel_x = 8 * self.vel_x / 9;
+                self.animate(3, 1, 2);
+
+                self.action_counter += 1;
+                if self.action_counter > 50 {
+                    self.cond.set_explode_die(true);
+                }
+            }
+            _ => (),
+        }
+
+        self.vel_y += 0x20;
+        if self.vel_y > 0x5ff {
+            self.vel_y = 0x5ff;
+        }
+
+        self.x += self.vel_x;
+        self.y += self.vel_y;
+
+        let dir_offset = if self.direction == Direction::Left { 0 } else { 3 };
+        self.anim_rect = state.constants.npc.n316_bute_dead[self.anim_num as usize + dir_offset];
+
+        Ok(())
+    }
+
+    pub(crate) fn tick_n323_bute_spinning(&mut self, state: &mut SharedGameState, players: [&mut Player; 2]) -> GameResult {
+
+        let player = self.get_closest_player_mut(players);
+    
+        self.animate(3, 0, 3);
+
+        match self.action_num {
+            0 | 1 => {
+                if self.action_num == 0 {
+                    self.action_num = 1;
+
+                    match self.direction {
+                        Direction::Left => {
+                            self.vel_x = -0x600;
+                        }
+                        Direction::Up => {
+                            self.vel_y = -0x600;
+                        }
+                        Direction::Right => {
+                            self.vel_x = 0x600;
+                        }
+                        Direction::Bottom => {
+                            self.vel_y = 0x600;
+                        }
+                        Direction::FacingPlayer => unreachable!(),
+                    }
+                }
+
+                self.action_counter += 1;
+                if self.action_counter == 16 {self.npc_flags.set_ignore_solidity(false)};
+
+                self.x += self.vel_x;
+                self.y += self.vel_y;
+
+                if self.flags.0 != 0 {self.action_num = 10};
```

Comment:
Issue I noticed with the bute arrow but never went back to fix it elsewhere. It should be `hit_anything()` instead of `any_flag()` since vanilla only checks for `flag & 0xFF`. Went through to fix it for other NPC projectiles that had the same logic.